### PR TITLE
Fix caching for Content Versioning

### LINKED ISSUE #20133: saving a version doesn't trigger cache auto purge
### Describe the Bug

saving a version:
image

refreshes the edit item form with cached data

setting `CACHE_ENABLED=false` fixes the issue

but... shouldn't the app always use the most up to date data, bypassing the cache? at least it should be an option i think...

### To Reproduce

# 1
use these directus cache settings:

```env
CACHE_ENABLED=true
CACHE_AUTO_PURGE=true
```

# 2
try to save a version

### Directus Version

v10.7.0

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by azrikahar**
## Scope

What's changed:

- Ensured cache value contains content version values when `version` query parameter is used
  Previously the `version` query param augments the payload _after_ `setCacheValue` is called, so the cached data is actually incorrect. This PR ensures the augmentation happens _before_ the cache is saved.
- Added auto cache purging for Content Version saves
  Uses the same code over here:
  https://github.com/directus/directus/blob/d9736d001c3b3cfede1e49e9549e6ee04a4decff/api/src/services/utils.ts#L129-L134
  
  which was added in #19115.

## Potential Risks / Drawbacks

- N/A at the moment

## Review Notes / Questions

- Partially unrelated, but should we mention both Content Version saves and also util sort (by #19115) in the description here:

    https://github.com/directus/directus/blob/b0c166ff46bee84f551864b3dc86796c06678b35/docs/self-hosted/config-options.md?plain=1#L483

---

Fixes #20133

### CODE REVIEW

**Comment by paescuj** [comment]
> Partially unrelated, but should we mention both Content Version saves and also util sort (by https://github.com/directus/directus/pull/19115) in the description here:

Or simply rephrase the sentence more generally. Could you do that right in this PR? 🙏

**Comment by azrikahar** [comment]
> Or simply rephrase the sentence more generally. Could you do that right in this PR? 🙏

Updated via https://github.com/directus/directus/pull/20153/commits/476a62d7aba29dd46998b0aebfd243f882cb64e9 👍

File: docs/self-hosted/config-options.md [hunk 1]
```diff
@@ -480,7 +480,7 @@ than you would cache database content. To learn more, see [Assets](#assets).
 | `CACHE_ENABLED`                              | Whether or not data caching is enabled.                                                                                 | `false`                              |
 | `CACHE_TTL`<sup>[1]</sup>                    | How long the data cache is persisted.                                                                                   | `5m`                                 |
 | `CACHE_CONTROL_S_MAXAGE`                     | Whether to not to add the `s-maxage` expiration flag. Set to a number for a custom value.                               | `0`                                  |
-| `CACHE_AUTO_PURGE`<sup>[2]</sup>             | Automatically purge the data cache on `create`, `update`, and `delete` actions.                                         | `false`                              |
+| `CACHE_AUTO_PURGE`<sup>[2]</sup>             | Automatically purge the data cache on `create`/`update`/`delete` actions, Content Version saves, and the sort utility.  | `false`                              |
```
Comment (paescuj) [inline docs/self-hosted/config-options.md:483]: I thought about a simpler sentence like this

```suggestion
| `CACHE_AUTO_PURGE`<sup>[2]</sup>             | Automatically purge the data cache on actions that manipulate the data.  | `false`                              |
```

WDYT?

Comment (azrikahar) [reply docs/self-hosted/config-options.md:483]:
```suggestion
| `CACHE_AUTO_PURGE`<sup>[2]</sup>             | Automatically purge the data cache on actions that manipulate the data.                                                 | `false`                              |
```

Sounds good 👍 will update it via this suggestion with formatted spacing.

Comment (paescuj) [reply docs/self-hosted/config-options.md]: Thankssss 🐍 ❤️

**Review by paescuj — APPROVED** [review]
Great work 🥇 Looks & works fine!

**Pull request merged**